[Hyperdrive] Hyperdrive pool placement improvement + ip address egress (#20239)
cloudflare/cloudflare-docs: src/content/changelog/hyperdrive/2025-03-04-hyperdrive-pooling-near-database-and-ip-range-egress.mdx

---

> title: Hyperdrive reduces query latency by up to 90% and now supports IP ACLs

Author: ```suggestion
title: Hyperdrive reduces query latency by up to 90% and now supports IP access control lists
```

> title: Hyperdrive reduces query latency by up to 90%, uses Cloudflare's IP addresses for egress

Reviewer: ```suggestion
title: Hyperdrive reduces query latency by up to 90% and now supports IP ACLs
```

> description: Hyperdrive now pools database connections closer to your database to reduce latency and uses Cloudflare's IP address ranges.

Reviewer: ```suggestion
description: Hyperdrive now pools database connections closer to your database to reduce query latency, and now also provides a documented IP address range for IP ACLs.
```

> description: Hyperdrive now pools database connections closer to your database to reduce latency and connects using Cloudflare's IP address ranges.

Reviewer 2: ```suggestion
description: Hyperdrive now pools database connections closer to your database to reduce latency and uses Cloudflare's IP address ranges.
```
I prefer this wording, if you feel that it accurately/clearly implies "use" == "use to connect"?

products:
  - hyperdrive

> date: 2025-03-06T18:00:00Z

Author: ```suggestion
date: 2025-03-07T18:00:00Z
```

---

Hyperdrive now pools database connections in one or more regions close to your database. This means that your uncached queries and new database connections have up to 90% less latency as measured from connection pools.

![Hyperdrive query latency decreases by 90% during Hyperdrive's gradual rollout of regional pooling.](~/assets/images/hyperdrive/configuration/hyperdrive-regional-pooling-query-latency-improvement.png)

> By improving placement of Hyperdrive database connection pools, Workers smart placement is now more effective when used with Hyperdrive, ensuring that your Worker can be placed as close to your database as possible.

Reviewer 2: ```suggestion
By improving placement of Hyperdrive database connection pools, Workers' Smart Placement is now more effective when used with Hyperdrive, ensuring that your Worker can be placed as close to your database as possible.
```

> With this update, Hyperdrive also uses [Cloudflare's standard IP address ranges](https://www.cloudflare.com/ips/) to connect to your database. This enables you to configure the firewall policies of your database to allow access to this limited IP address range.

Reviewer: ```suggestion
With this update, Hyperdrive also uses [Cloudflare's standard IP address ranges](https://www.cloudflare.com/ips/) to connect to your database. This enables you to configure the firewall policies (IP access control lists) of your database to only allow access from Cloudflare and Hyperdrive.
```
